src/shared/sleep-config.c

```diff
@@ -196,25 +197,31 @@ static int read_battery_capacity_percentage(sd_device *dev) {
         return battery_capacity;
 }
 
-/* If battery percentage capacity is <= 5%, return success */
-int battery_is_low(void) {
+/* If a battery whose percentage capacity is <= 5% exists, and we're not on AC power, return success */
+int battery_is_present_and_low(void) {
         _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
         sd_device *dev;
+        bool has_battery = false;
         int r;
 
          /* We have not used battery capacity_level since value is set to full
          * or Normal in case ACPI is not working properly. In case of no battery
          * 0 will be returned and system will be suspended for 1st cycle then hibernated */
 
+        if (on_ac_power())
```

Comment:
on_ac_power() might return error. right now you are treating an error like false. Which is OK I think, but we usually want this to be don explicitly, i.e. if we fold the error case into either true or false, then please make this visible by doing a literal check like `!= 0`, and maybe adding a comment.

(and maybe even doing log_debug_errno() in case of the error)